src/API/JSONData.hs

```diff
@@ -107,7 +107,7 @@ data SpielResponse =
   -- | represents a win/lose result
   SpielGameResult String |
   -- | represents a type error
-  SpielTypeError TypeError |
+  SpielTypeError Error  |
```

Comment:
Eventually, we could consolidate all the errors into a single `Error` response (if that is beneficial)

Reply:
Also, I am not much of a web developer / API guy, so I'm all ears concerning this stuff.

Reply:
I'm totally on board with that. From the web-end of things it would be great to differentiate quickly between success and failure. A general error is perfectly acceptable for the case that something went wrong, but only if we can capture the nature of all error types under that. Such that we have the type/runtime/parser errors and their associated code/message data that can all be represented equally. And since this is haskell I'm pretty sure we can just whip up a few data constructors for the type Error that do exactly that and we're done 👍 .

Reply:
To further go in that direction, it would make sense to consolidate all of these other error types listed below into a type of Error as well, we can talk about it. But it would make it more organizationally agreeable. Makes me think we could just do away with this wrapper and report the Errors directly as a parsable type, taking advantage of what the parser/typecheker/runtime reports instead of reformatting it.